azure/datalake/store/lib.py

```diff
@@ -216,18 +216,34 @@ def _check_token(self):
             self.token = refresh_token(self.token)
             self.head = {'Authorization': 'Bearer ' + self.token['access']}
 
-    def _log_request(self, method, url, headers):
-        logger.debug("Request Method: %s", method.upper())
-        logger.debug("Request URL: %s", url)
-        for header in headers:
-            logger.debug("Request Header: %s='%s'", header, headers[header])
+    def _log_request(self, method, url, op, path, params, headers):
+        msg = "HTTP Request\n{} {}\n".format(method.upper(), url)
+        msg += "{} '{}' {}\n\n".format(
+            op, path,
+            " ".join(["{}={}".format(key, params[key]) for key in params]))
+        msg += "\n".join(["{}: {}".format(header, headers[header])
+                          for header in headers])
+        logger.debug(msg)
 
     def _log_response(self, response, payload=False):
-        logger.debug("Response Status Code: %s", response.status_code)
-        for header in response.headers:
-            logger.debug("Response Header: %s='%s'", header, response.headers[header])
+        msg = "HTTP Response\n{}\n{}\n".format(
+            response.status_code,
+            "\n".join(["{}: {}".format(header, response.headers[header])
+                       for header in response.headers]))
         if payload:
-            logger.debug("Response Body: %s", pprint.pformat(response.content))
+            msg += "\n{}".format(pprint.pformat(response.content))
```

Comment:
be careful with this, since download (op=OPEN) has a payload that can be quite large. I would recommend protecting yourself here and do one of the following:
1. ignore the payload specifically for the OPEN op when it has a 200 response and write out that the output was ignored due to the method called
2. Have a blanket truncation rule where the payload is logged up to some amount of data (100KB, maybe?) with a message after indicating it was truncated.